src/Sylius/Bundle/ResourceBundle/Behat/DefaultContext.php

```diff
@@ -340,4 +321,109 @@ protected function fixStepArgument($argument)
     {
         return str_replace('\\"', '"', $argument);
     }
+
+    /**
+     * @param NodeElement $table
+     * @param string $columnName
+     *
+     * @return integer
+     *
+     * @throws \Exception If column was not found
+     */
+    protected function getColumnIndex(NodeElement $table, $columnName)
+    {
+        $rows = $table->findAll('css', 'tr');
+
+        if (!isset($rows[0])) {
+            throw new \Exception("There are no rows!");
+        }
+
+        /** @var NodeElement $firstRow */
+        $firstRow = $rows[0];
+        $columns = $firstRow->findAll('css', 'th,td');
+        foreach ($columns as $index => $column) {
+            /** @var NodeElement $column */
+            if (0 === stripos($column->getText(), $columnName)) {
+                return $index;
+            }
+        }
+
+        throw new \Exception(sprintf('Column with name "%s" not found!', $columnName));
+    }
+
+    /**
+     * @param NodeElement $table
+     * @param array $fields
+     *
+     * @return NodeElement|null
+     *
+     * @throws \Exception If column was not found
+     */
+    protected function getRowWithFields(NodeElement $table, array $fields)
+    {
+        $foundRows = $this->getRowsWithFields($table, $fields, true);
+
+        if (empty($foundRows)) {
+            return null;
+        }
+
+        return $foundRows;
+    }
+
+    /**
+     * @param NodeElement $table
+     * @param array $fields
+     * @param boolean $first
+     *
+     * @return NodeElement[]|NodeElement
+     *
+     * @throws \Exception If column was not found
+     */
+    protected function getRowsWithFields(NodeElement $table, array $fields, $first = false)
+    {
+        $rows = $table->findAll('css', 'tr');
+
+        if (!isset($rows[0])) {
+            throw new \Exception("There are no rows!");
+        }
+
+        foreach ($fields as $columnName => $expectedValue) {
+            $columnIndex = $this->getColumnIndex($table, $columnName);
+
+            $fields[$columnIndex] = $expectedValue;
+            unset($fields[$columnName]);
+        }
+
+        $foundRows = array();
+
+        /** @var NodeElement[] $rows */
+        $rows = $table->findAll('css', 'tr');
+        foreach ($rows as $row) {
+            $found = true;
+
+            /** @var NodeElement[] $columns */
+            $columns = $row->findAll('css', 'th,td');
+            foreach ($fields as $index => $searchedValue) {
+                if (!isset($columns[$index])) {
+                    throw new \InvalidArgumentException(sprintf('There is no column with index %d', $index));
+                }
+
+                if (0 !== stripos(trim($columns[$index]->getText()), trim($searchedValue))) {
+                    $found = false;
+
+                    break;
+                }
+            }
+
+            if ($found) {
+                $foundRows[] = $row;
+
+                if ($first) {
```

Comment:
Ok then this two steps (https://github.com/Sylius/Sylius/pull/3098/files#diff-444c7166df771289a30354d3c2e236acR359, https://github.com/Sylius/Sylius/pull/3098/files#diff-444c7166df771289a30354d3c2e236acR373) looks only for the first row?